obnoxious.py

```diff
@@ -1,67 +1,130 @@
+    while True:
+        choice = input("Choose the action: ")
 
-        if not greedy_success:
-            print("The Minimum Hazard Facility Greedy failed.")
-            return
+        start = 0
+        end = 0
 
-    if greedy == 2:
+        iterations = 0
+        improvements = 0
 
-        start = time.time()
-        greedy_success = mchrf_greedy(facilities, towns)
-        end = time.time()
+        if choice == "exit":
+            print("\nGoodbye!\n")
+            break
 
-        if not greedy_success:
-            print("The Minimum Capacity/Hazard Ratio Facility Greedy failed.")
-            return
+        if (choice == "3" or choice == "4" or choice == "5" or choice == "6" or choice == "7") and not facilities:
+            print("\nThere are no instance set up to work on; you need to give one first.\n")
+            continue
 
-    duration = end - start
+        if choice == "1":
 
-    show_greedy_result(facilities, towns, hazards, duration)
+            num_facilities = int(input("Insert the number of facilities: "))
+            num_towns = int(input("Insert the number of towns: "))
+            filename = input("Insert the file name: ")
+            generate_instance(num_facilities, num_towns, filename)
 
-    if local_search == 1:
+            print(f"\nNew instance '{filename}' with {num_facilities} facilities and {num_towns} towns has been "
+                  f"generated in './instances'\n")
 
-        start = time.time()
-        iterations, improvements = large_neighborhood_search(facilities, towns, hazards, 50, 10)
-        end = time.time()
+        if choice == "2":
 
-    if local_search == 2:
+            filename = input("Insert instance file relative path: ")
 
-        start = time.time()
-        iterations, improvements = variable_depth_search(facilities, towns, hazards, 50, 10)
-        end = time.time()
+            path = pathlib.Path(f"./instances/{filename}.json")
 
-    duration = end - start
+            if path.exists() and path.is_file():
+                facilities, towns, hazards = setup(f"./instances/{filename}.json")
 
-    show_local_search_result(facilities, towns, hazards, duration, iterations, improvements)
+                print("\nInstance setup completed correctly\n")
+            else:
+                print("\nFile doesn't exist\n")
 
-    # show_facility_usage_by_town(towns)
+        if choice == "3":
+
+            start = time.time()
+            greedy_success = mhf_greedy(facilities, towns)
+            end = time.time()
+
+            if not greedy_success:
+                print("The Minimum Hazard Facility Greedy failed.")
+                continue
+
+        if choice == "4":
+
+            start = time.time()
+            greedy_success = mchrf_greedy(facilities, towns)
+            end = time.time()
+
+            if not greedy_success:
+                print("The Minimum Capacity/Hazard Ratio Facility Greedy failed.")
+                continue
+

```

Comment:
Ci vuole un controllo su entrambe le local serach che non le faccia eseguire se prima non ho eseguito una greedy.
Altrimenti quando viene chiamata su un'istanza con tutte le facility chiuse va in crash.
Direi che basta inserire un booleano che vhe viene inizializzato a False fuori dal while, mentre viene messo a True dopo che una greedy ha terminato l'esecuzione.